### PR TITLE
session: change the global variable tidb_stmt_summary_max_stmt_count value from 200 to 3000 when upgrade

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

PR https://github.com/pingcap/tidb/pull/25843 only changes the default value, but for the TiDB cluster which upgrades from the old version, the global variable `tidb_stmt_summary_max_stmt_count` value still is old value `200`.

This PR changes the global variable tidb_stmt_summary_max_stmt_count value to 3000 if the old value is small than 3000 when upgrade.

### Check List 

Tests 

- [x] Unit test
- [x] Manual test

### Release note 

```release-note
None.
```

### CODE REVIEW

**Comment by morgo** [comment]
LGTM, but I think this will be hard to cherry pick :(

Bootstrap task numbers are universal. So a 4.0 server would need to execute up to task `73` to apply this. Renumbering tasks is also difficult because it can mean some bootstrap items are missed after upgrade.

If you must cherry-pick it, the required way to do it is to 'merge' bootstrap numbers in earlier versions. This is what I did in https://github.com/pingcap/tidb/pull/25750 -- but it is messy.

**Comment by crazycs520** [comment]
> LGTM, but I think this will be hard to cherry-pick :(
> 
> Bootstrap task numbers are universal. So a 4.0 server would need to execute up to task `73` to apply this. Renumbering tasks is also difficult because it can mean some bootstrap items are missed after upgrade.
> 
> If you must cherry-pick it, the required way to do it is to 'merge' bootstrap numbers in earlier versions. This is what I did in #25750 -- but it is messy.

After consideration, I found It's really hard to cherry-pick...

**Comment by crazycs520** [comment]
@morgo @tiancaiamao @djshow832  PTAL.
I don't plan to do cherry-pick anymore.

File: session/bootstrap.go [hunk 1]
```diff
@@ -1546,6 +1549,13 @@ func upgradeToVer73(s Session, ver int64) {
 	doReentrantDDL(s, CreateCapturePlanBaselinesBlacklist)
 }
 
+func upgradeToVer74(s Session, ver int64) {
```
Comment (tiancaiamao) [inline session/bootstrap.go:1552]: Do we need to update the bootstrap branch?
When it's bootstrapped from a new cluster is the default value correct?

**Comment by tiancaiamao** [comment]
And by the way, why this is both a configure item and a global variable?

**Comment by morgo** [comment]
LGTM, but awaiting an answer on @tiancaiamao's question. The behavior of most "instance" config options is Session, but the behavior of `tidb_stmt_summary_max_stmt_count` might be weird since it is both session + global.

File: session/bootstrap.go [hunk 2]
```diff
@@ -1546,6 +1549,13 @@ func upgradeToVer73(s Session, ver int64) {
...
+	if ver >= version74 {
+		return
+	}
+	mustExecute(s, fmt.Sprintf("UPDATE mysql.global_variables SET VARIABLE_VALUE='%[1]v' WHERE VARIABLE_NAME = 'tidb_stmt_summary_max_stmt_count' AND CAST(VARIABLE_VALUE AS SIGNED) < %[1]v", config.GetGlobalConfig().StmtSummary.MaxStmtCount))
```
Comment (djshow832) [inline session/bootstrap.go:1556]: I don't think so. What if the user has a custom value, say, 500? It's weird to change that.

Comment (crazycs520) [reply session/bootstrap.go:1556]: Yes, but It's not bad that change from 500 to 3000?

Comment (xhebox) [reply session/bootstrap.go:1556]: Is it acceptable to only upgrades from `200`?

Comment (crazycs520) [reply session/bootstrap.go:1556]: I still prefer to enlarge this value when the value is small than 200.  With the bigger capacity of `statements_summary`, it is much more useful in troubleshooting.

Comment (crazycs520) [reply session/bootstrap.go:1556]: ok

Comment (crazycs520) [reply session/bootstrap.go]: @xhebox PTAL

Comment (xhebox) [reply session/bootstrap.go]: I remembered you have replied: I still prefer to enlarge the value. Whatever, this has been resolved.

**Review by xhebox — APPROVED** [review]

**Pull request merged**